model/src/main/kotlin/VulnerabilityReference.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import java.net.URI
+
+/**
+ * A data class representing detailed information about a vulnerability obtained from a specific source.
+ *
+ * A single vulnerability can be listed by multiple sources using different scoring systems to denote its severity.
+ * So when ORT queries different providers for vulnerability information it may well find multiple records for a single
+ * vulnerability, which could even contain contradicting information. To model this, a [Vulnerability] is associated
+ * with a list of references; each reference points to the source of the information and has some detail information
+ * provided by this source.
+ */
+data class VulnerabilityReference(
+    /**
+     * The URI pointing to details of this vulnerability. This can also be used to derive the source of this
+     * information.
+     */
+    val url: URI,
+
+    /**
+     * The name of the scoring system to express the severity of this vulnerability if available.
+     */
+    val scoringSystem: String?,
```

Comment:
This does not need to be done in this PR, but I wonder if this should be rather an enum class in the future. As you already mentioned somewhere in this PR, `NexusIq` specifically provides two scoring systems, i.e., `CVSS2` and `CVSS3`. Those seem to be official scoring systems.

Reply:
Agreed, would be good to have. This could also help with the interpretation of these results, e.g. to compare values from different scoring systems. However, I am not sure whether we have sufficient information yet to model such an enumeration.